lib/pacemaker/pcmk_scheduler.c

```diff
@@ -759,6 +759,65 @@ unpack_cib(xmlNode *cib, unsigned long long flags, pe_working_set_t *data_set)
     cluster_status(data_set); // Sets pe_flag_have_status
 }
 
+/*!
+ * \internal
+ * \brief Does a given start action have a matching pseudo-start action for
+ *        a remote resource?
+ *
+ * \param[in] data_set     Cluster working set
+ * \param[in] start_action The start action to find a match for
+ */
+static bool
+has_remote_pseudo_stop_action(const pe_working_set_t *data_set,
+                              const pe_action_t *start_action)
+{
+    for (GList *iter = data_set->actions; iter != NULL; iter = iter->next) {
+        pe_action_t *action = (pe_action_t *) iter->data;
+
+        if (pcmk_is_set(action->flags, pe_action_pseudo) &&
```

Comment:
We probably need to also check ! pe_action_optional

Reply:
Added.